specification/security/resource-manager/Microsoft.Security/preview/2024-07-01-preview/securityConnectors.json

```diff
@@ -0,0 +1,1707 @@
+        },
+        "managementProjectNumber": {
+          "type": "string",
+          "description": "The GCP management project number from organizational onboarding"
+        }
+      }
+    },
+    "GithubScopeEnvironmentData": {
+      "type": "object",
+      "description": "The github scope connector's environment data",
+      "x-ms-discriminator-value": "GithubScope",
+      "allOf": [
+        {
+          "$ref": "#/definitions/EnvironmentData"
+        }
+      ],
+      "properties": {}
+    },
+    "AzureDevOpsScopeEnvironmentData": {
+      "type": "object",
+      "description": "The AzureDevOps scope connector's environment data",
+      "x-ms-discriminator-value": "AzureDevOpsScope",
+      "allOf": [
+        {
+          "$ref": "#/definitions/EnvironmentData"
+        }
+      ],
+      "properties": {}
+    },
+    "GitlabScopeEnvironmentData": {
+      "type": "object",
+      "description": "The GitLab scope connector's environment data",
+      "x-ms-discriminator-value": "GitlabScope",
+      "allOf": [
+        {
+          "$ref": "#/definitions/EnvironmentData"
+        }
+      ],
+      "properties": {}
+    },
+    "DockerHubEnvironmentData": {
+      "type": "object",
+      "description": "The Docker Hub connector environment data",
+      "x-ms-discriminator-value": "DockerHubOrganization",
+      "allOf": [
+        {
+          "$ref": "#/definitions/EnvironmentData"
+        }
+      ],
+      "properties": {
+        "authentication": {
+          "type": "object",
+          "description": "The Docker Hub organization authentication details",
+          "$ref": "#/definitions/Authentication"
+        },
+        "scanInterval": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Scan interval in hours (value should be between 1-hour to 24-hours)"
+        }
+      }
+    },
+    "Authentication": {
+      "type": "object",
+      "description": "The environment authentication details",
+      "discriminator": "authenticationType",
+      "required": [
+        "authenticationType"
+      ],
+      "properties": {
+        "authenticationType": {
+          "type": "string",
+          "description": "The authentication type",
+          "enum": [
+            "AccessToken"
+          ],
+          "x-ms-enum": {
+            "name": "authenticationType",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "AccessTokenAuthentication": {
+      "type": "object",
+      "description": "The environment authentication details",
+      "x-ms-discriminator-value": "AccessToken",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Authentication"
+        }
+      ],
+      "properties": {
+        "username": {
+          "type": "string",
+          "description": "The user name that will be used while authenticating with the onboarded environment"
+        },
+        "accessToken": {
```

Comment:
I wonder why this is needed ? is this being shown to the user ? This is an access related information and should be secret right ?!

Reply:
This version is build on top of the older version with the goal of supporting a new environment only, without the intention to make further unrelated changes